tests/pkg/clients/destination_client.go

```diff
@@ -114,3 +117,43 @@ func (c *DestinationClient) DeleteDestination(t *testing.T, destinationName stri
 	require.NoError(t, err)
 	require.Equal(t, http.StatusOK, resp.StatusCode)
 }
+
+func (c *DestinationClient) GetDestinationByName(t *testing.T, destinationName string, expectedStatusCode int) json.RawMessage {
+	url := c.apiURL + c.apiConfig.EndpointTenantDestinations + "/" + url.QueryEscape(destinationName)
+	request, err := http.NewRequest(http.MethodGet, url, nil)
+	require.NoError(t, err)
+
+	resp, err := c.httpClient.Do(request)
+	require.NoError(t, err)
```

Comment:
Actually, the http package and the `Do` method take care to close the resp body in case of an error, plus the response will be nil in that case.
Moreover, we need explicitly to check that the resp is not nil + the `resp.Body` is not nil and after that, we could execute `resp.Body.Close()` in the defer.